### PR TITLE
Proper property escaping

### PR DESCRIPTION
**Pull request by keszybz**
From https://github.com/systemd/systemd/issues/6254#issuecomment-312452050:
```console
$ systemctl list-dependencies 'dev-mapper-luks\x2d8db85dcf\x2d6230\x2d4e88\x2d940d\x2dba176d062b31.device'
dev-mapper-luks\x2d8db85dcf\x2d6230\x2d4e88\x2d940d\x2dba176d062b31.device
● ├─dev-mapper-luks\x2d8db85dcf\x2d6230\x2d4e88\x2d940d\x2dba176d062b31.swap
● └─systemd-cryptsetup@luks\x2d8db85dcf\x2d6230\x2d4e88\x2d940d\x2dba176d062b31.service
$ systemctl show -p Wants,Requires,RequiredBy,WantedBy,PartOf 'dev-mapper-luks\x2d8db85dcf\x2d6230\x2d4e88\x2d940d\x2dba176d062b31.device'
Requires=systemd-cryptsetup@luks\x5cx2d8db85dcf\x5cx2d6230\x5cx2d4e88\x5cx2d940d\x5cx2dba176d062b31.service
Wants=dev-mapper-luks\x5cx2d8db85dcf\x5cx2d6230\x5cx2d4e88\x5cx2d940d\x5cx2dba176d062b31.swap
PartOf=
RequiredBy=
WantedBy=
```

### CODE REVIEW

File: src/shared/bus-util.c [hunk 1]
```diff
@@ -724,8 +724,12 @@ int bus_print_property(const char *name, sd_bus_message *property, bool value, b
                 if (r < 0)
                         return r;
 
-                if (all || !isempty(s))
-                        print_property(name, "%s", s);
+                if (all || !isempty(s)) {
+                        bool good;
+
+                        good = !strchr(s, '\n');
```
Comment (poettering) [inline src/shared/bus-util.c]: I think there should be a comment here, explaining why we look for `\n` here, and nothing else (for example, `=`?).

Also maybe turn the check into a static inline mini-function?

File: src/shared/bus-util.c [hunk 2]
```diff
@@ -845,10 +849,14 @@ int bus_print_property(const char *name, sd_bus_message *property, bool value, b
                                 return r;
 
                         while ((r = sd_bus_message_read_basic(property, SD_BUS_TYPE_STRING, &str)) > 0) {
+                                bool good;
+
                                 if (first && !value)
                                         printf("%s=", name);
 
-                                printf("%s%s", first ? "" : " ", str);
+                                good = str[strcspn(str, " \n")] == '\0';
```
Comment (poettering) [inline src/shared/bus-util.c]: as above

**Comment by keszybz** [comment]
I think an inline function would be overkill (it's a single line after all).

Updated with comments.

**Comment by poettering** [comment]
lgtm

**Comment by poettering** [comment]
centos CI fails due to the missing BTN definition, which was fixed in the meantime. Merging hence.

**Pull request merged**